src/apps/workspaces/crud/user_applet_access.py

```diff
@@ -618,16 +618,76 @@ async def get_workspace_managers(
             .correlate(AppletSchema)
         )
 
-        query: Query = (
+        invited_users = (
+            select(
+                InvitationSchema.id,
+                InvitationSchema.first_name,
+                InvitationSchema.last_name,
+                InvitationSchema.email,
+                func.array_agg(
+                    aggregate_order_by(
+                        func.distinct(InvitationSchema.title),
+                        InvitationSchema.title,
+                    )
+                ).label("titles"),
+                func.array_agg(
+                    aggregate_order_by(
+                        func.distinct(InvitationSchema.role),
+                        InvitationSchema.role,
+                    )
+                ).label("roles"),
+                func.array_agg(
+                    aggregate_order_by(
+                        func.json_build_object(
+                            text("'applet_id'"),
+                            AppletSchema.id,
+                            text("'applet_display_name'"),
+                            AppletSchema.display_name,  # noqa: E501
+                            text("'applet_image'"),
+                            AppletSchema.image,
+                            text("'role'"),
+                            InvitationSchema.role,
+                            text("'encryption'"),
+                            AppletSchema.encryption,
+                        ),
+                        AppletSchema.id,
+                    )
+                ).label("applets"),
+                InvitationSchema.status,
+                InvitationSchema.created_at,
+                InvitationSchema.user_id,
+            )
+            .select_from(InvitationSchema)
+            .join(
+                AppletSchema,
+                and_(
+                    AppletSchema.id == InvitationSchema.applet_id,
+                    AppletSchema.soft_exists(),
+                ),
+            )
+            .where(
+                InvitationSchema.role != Role.RESPONDENT,
+                InvitationSchema.status == InvitationStatus.PENDING,
+                InvitationSchema.applet_id == applet_id if applet_id else True,
+            )
+            .group_by(InvitationSchema.id)
```

Comment:
Looks redundant as it's always one record in group

Reply:
yeah, but I'm keep same returning format as `accepted_users` query, in this case roles and titles are arrays.